src/sentry/sentry_metrics/querying/samples_list.py

```diff
@@ -84,10 +84,62 @@ def get_spans_by_key(self, span_ids: list[tuple[str, str, str]]):
         return builder.process_results(query_results)
 
 
+class SegmentsSamplesListExecutor(SamplesListExecutor):
+    @classmethod
+    def mri_to_column(cls, mri) -> str | None:
+        if mri == TransactionMRI.DURATION.value:
+            return "duration"
+        return None
+
+    @classmethod
+    def supports(cls, mri: str) -> bool:
+        return cls.mri_to_column(mri) is not None
+
+    def execute(self, offset, limit):
+        span_keys = self.get_span_keys(offset, limit)
+        return self.get_spans_by_key(span_keys)
+
+    def get_span_keys(self, offset: int, limit: int) -> list[tuple[str, str, str]]:
+        rounded_timestamp = f"rounded_timestamp({self.rollup})"
+
+        builder = QueryBuilder(
```

Comment:
I was wondering, any specific reason why we are using complex query builders for queries as simple as the ones we are running?

Reply:
The query builders allow us to have a uniform entry point to the various datasets as well as correctly translate user queries to the underlying table.

For example, a query like `transaction:foo` needs to be translated differently depending on the dataset we're looking at. On the metrics tables, it's a tag, on the spans table it's the segment name column. Using the query builders, it handles these compatibility issues by translating the search according to the underlying dataset.

I do understand why the query builder may seem complex at first but much of its complexity has grown out of necessity to handle these edge cases. And simplifying some of this complexity is something we want to tackle in the future.

Reply:
Yeah, I am aware of that. It has noticeable benefits if you are querying a broad range of things but debugging any problem can become quite tricky due to the many layers of interaction. I am fine with using it but I was hoping we could build simple queries on our own given that they would be mostly static.

Reply:
To my understanding, one of the things we want out of this sample list is the ability to filter on various fields and tags. So much of that will be handled by the query builder.

I've already run into issues with the first iteration from where searching on a transaction name means the query no longer works because `transaction` is not a valid column in the spans dataset.

Reply:
It was mapped accordingly here: https://github.com/getsentry/sentry/blob/9e1708439e9010dd3d523adb112f0d5da7f966c4/src/sentry/sentry_metrics/querying/metadata/metrics_correlations.py#L62-L62

Reply:
But yeah, that's a big benefit. Especially if we want to query different datasets with their own configs.